### PR TITLE
Facet config constants

### PR DESCRIPTION
**Pull request by aslagle**
This adds default and enabled facets to the library-wide configuration settings.

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -145,6 +152,24 @@ class Configuration(object):
...
         },
+    ] + [
+        { "key": ENABLED_FACETS_KEY_PREFIX + group,
+          "label": _("Enabled %(group)s facets", group=display_name),
+          "type": "list",
+          "options": [
+              { "key": facet, "label": FacetConstants.FACET_DISPLAY_TITLES.get(facet) }
+              for facet in FacetConstants.FACETS_BY_GROUP.get(group)
+          ],
+        } for group, display_name in FacetConstants.GROUP_DISPLAY_TITLES.iteritems()
```
Comment (aslagle) [inline config.py]: In the admin interface, this shows up with a heading for each group and a checkbox for each facet in the group.

File: config.py [hunk 2]
```diff
@@ -145,6 +152,24 @@ class Configuration(object):
...
+    ] + [
+        { "key": DEFAULT_FACET_KEY_PREFIX + group,
+          "label": _("Default %(group)s facet", group=display_name),
+          "type": "select",
+          "options": [
+              { "key": facet, "label": FacetConstants.FACET_DISPLAY_TITLES.get(facet) }
+              for facet in FacetConstants.FACETS_BY_GROUP.get(group)
+          ],
+        } for group, display_name in FacetConstants.GROUP_DISPLAY_TITLES.iteritems()
```
Comment (aslagle) [inline config.py]: This shows up as a dropdown for each group. There's currently no validation that the default facet is one of the enabled facets. I'm not sure if that's important.

Comment (leonardr) [inline config.py]: It's not important right now; almost nobody will configure these settings.

File: facets.py [hunk 3]
```diff
@@ -21,10 +31,25 @@ class FacetConstants(object):
...
     ORDER_RANDOM = 'random'
+    ORDER_FACETS = [
+        ORDER_TITLE,
+        ORDER_AUTHOR,
+        ORDER_LAST_UPDATE,
+        ORDER_ADDED_TO_COLLECTION,
+        ORDER_SERIES_POSITION,
+        ORDER_WORK_ID,
+        ORDER_RANDOM,
+    ]
```
Comment (leonardr) [inline facets.py]: Some of these facets only make sense in certain contexts and shouldn't be options in the user interface, which controls how lane feeds are presented to patrons. "random" might be fun, but I don't think there's any reason to include "last update" or "work id", and series position won't work at all.

Comment (aslagle) [inline facets.py]: Yeah, I wasn't sure about those. I'll remove them from this list I guess.

**Comment by leonardr** [comment]
:+1:

**Pull request merged**